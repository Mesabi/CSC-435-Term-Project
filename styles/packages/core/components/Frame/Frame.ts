import styled, { system, SystemProps } from '@xstyled/styled-components';

export interface FrameProps extends SystemProps {}

const Frame = styled.div<FrameProps>(system);

Frame.defaultProps = {
  w: 'auto',
  h: 'auto',
  bg: 'material',
  boxShadow: 'out',
};

export default Frame;
