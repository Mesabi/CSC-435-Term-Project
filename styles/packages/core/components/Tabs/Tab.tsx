import React, { forwardRef } from 'react';
import styled, { css, th } from '@xstyled/styled-components';

const NavItem = styled.li<{ active: boolean }>`
  list-style: none;
  padding: 3 6;
  background-color: material;
  box-shadow: inset 0 1px 0 0 ${th('colors.borderLightest')},
    -2px 1px 0 -1px ${th('colors.material')},
    -3px 1px 0 -1px ${th('colors.borderLightest')},
    -2px 0 0 -1px ${th('colors.borderLightest')},
    2px 1px 0 -1px ${th('colors.borderDark')},
    2px 0 0 -1px ${th('colors.borderDarkest')},
    3px 1px 0 -1px ${th('colors.borderDarkest')};
  margin-right: 2;
  margin-left: 2;

  &:first-child {
    margin-left: 4;
  }

  ${({ active }) =>
    active &&
    css`
      margin-top: -2;
      margin-left: -2;
      margin-bottom: 1;

      &:first-child {
        margin-left: 2;
      }

      box-shadow: inset 0 1px 0 0 ${th('colors.borderLightest')},
        -2px 1px 0 -1px ${th('colors.material')},
        -0.5px 3px 0 0.5px ${th('colors.material')},
        -2px 2px 0 0 ${th('colors.borderLightest')},
        -2px 0 0 -1px ${th('colors.borderLightest')},
        1px 2px 0 0 ${th('colors.borderDark')},
        2px 0 0 -1px ${th('colors.borderDarkest')},
        2px 2px 0 0 ${th('colors.borderDarkest')};

      & + li {
        margin-left: 0;
        box-shadow: inset 0 1px 0 0 ${th('colors.borderLightest')},
          2px 1px 0 -1px ${th('colors.borderDark')},
          2px 0 0 -1px ${th('colors.borderDarkest')},
          3px 1px 0 -1px ${th('colors.borderDarkest')};
      }
    `}
`;

export interface TabProps extends React.HTMLAttributes<HTMLLIElement> {
  activeTab?: string;
  title: string;
  disabled?: boolean;
  onClick?(e: React.MouseEvent): void;
}

const Tab = forwardRef<HTMLLIElement, TabProps>(
  ({ activeTab, title, ...rest }, ref) => (
    <NavItem {...rest} active={activeTab === title} ref={ref}>
      {title}
    </NavItem>
  ),
);

export default Tab;
