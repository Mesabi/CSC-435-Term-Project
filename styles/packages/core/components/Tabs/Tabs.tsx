import React, { forwardRef } from 'react';
import styled from '@xstyled/styled-components';

import Tab, { TabProps } from './Tab';
import Frame from '../Frame';

const Navbar = styled(Frame)`
  display: flex;
  padding: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: unset;
`;

const NavContainer = styled(Frame)`
  padding: 12;
  box-shadow: out;
`;

NavContainer.displayName = 'NavContainer';
interface TabsProps {
  style?: React.CSSProperties;
  defaultActiveTab?: string;
  children: React.ReactElement<TabProps> | Array<React.ReactElement<TabProps>>;

  onChange?(title: string, e: React.MouseEvent): void;
}

const Tabs = forwardRef<HTMLOListElement, TabsProps>(
  ({ children, style, defaultActiveTab, onChange, ...rest }, ref) => {
    const [firstTab] = React.Children.toArray(children) as Array<
      React.ReactElement<TabProps>
    >;
    const [activeTab, setActiveTab] = React.useState(
      defaultActiveTab || firstTab.props.title,
    );

    return (
      <>
        <Navbar style={style} {...rest} as="ol" ref={ref}>
          {React.Children.map(
            children,
            (child: React.ReactElement<TabProps>) => {
              const { title, disabled } = child.props;

              return (
                <Tab
                  key={title}
                  {...child.props}
                  activeTab={activeTab}
                  onClick={e => {
                    if (!disabled) {
                      if (onChange) {
                        onChange(title, e);
                      }
                      setActiveTab(title);
                    }
                  }}
                />
              );
            },
          )}
        </Navbar>

        <NavContainer style={style}>
          {React.Children.map(
            children,
            (child: React.ReactElement<TabProps>) =>
              child.props.title === activeTab && child.props.children,
          )}
        </NavContainer>
      </>
    );
  },
);

export default Tabs;
