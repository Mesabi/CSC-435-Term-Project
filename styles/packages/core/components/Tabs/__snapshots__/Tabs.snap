// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> Snapshot should match snapshot 1`] = `
.c0 {
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c2 {
  list-style: none;
  padding: 3px 6px;
  background-color: #c3c7cb;
  box-shadow: inset 0 1px 0 0 #ffffff, -2px 1px 0 -1px #c3c7cb, -3px 1px 0 -1px #ffffff, -2px 0 0 -1px #ffffff, 2px 1px 0 -1px #868a8e, 2px 0 0 -1px #000000, 3px 1px 0 -1px #000000;
  margin-right: 2px;
  margin-left: 2px;
  margin-top: -2px;
  margin-left: -2px;
  margin-bottom: 1px;
  box-shadow: inset 0 1px 0 0 #ffffff, -2px 1px 0 -1px #c3c7cb, -0.5px 3px 0 0.5px #c3c7cb, -2px 2px 0 0 #ffffff, -2px 0 0 -1px #ffffff, 1px 2px 0 0 #868a8e, 2px 0 0 -1px #000000, 2px 2px 0 0 #000000;
}

.c2:first-child {
  margin-left: 4px;
}

.c2:first-child {
  margin-left: 2px;
}

.c2 + li {
  margin-left: 0;
  box-shadow: inset 0 1px 0 0 #ffffff, 2px 1px 0 -1px #868a8e, 2px 0 0 -1px #000000, 3px 1px 0 -1px #000000;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: unset;
}

.c3 {
  padding: 12px;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

<div>
  <ol
    class="c0 c1"
  >
    <li
      class="c2"
    >
      example
    </li>
  </ol>
  <div
    class="c0 c3"
  >
    <p>
      Example text
    </p>
  </div>
</div>
`;
